android-pay/src/main/java/com/stripe/wrap/pay/activity/StripeAndroidPayActivity.java

```diff
@@ -427,32 +425,28 @@ protected void onFullWalletRetrieved(@Nullable FullWallet fullWallet) {
             Log.w(TAG, "Null token returned with non-null full wallet");
         }
 
-        try {
-            Token token = TokenParser.parseToken(rawPurchaseToken);
-            if (AndroidPayConfiguration.getInstance().getUsesSources()) {
-                getStripeSource(fullWallet, token);
-            } else {
-                logApiCallOnNewThread(token, null);
-                onStripePaymentSourceReturned(fullWallet, token);
-            }
-        } catch (JSONException jsonException) {
-            Log.i(TAG,
-                    String.format(Locale.ENGLISH,
-                            "Could not parse object as Stripe token. Trying as Source.\n%s",
-                            rawPurchaseToken),
-                    jsonException);
+        Token token = Token.fromString(rawPurchaseToken);
+        if (token == null) {
+            // Could not parse object as Token, trying as Source
```

Comment:
Noting that the Log.i message for could not parse object is now a comment. No opinions just wanted to make sure it's intentional.